### PR TITLE
Recipe App (RoR Group Project)

### PR DESCRIPTION
**Pull request by pratap-panabaka**
## Task

The app will have some common interfaces, but depending on your team size it will have a couple of extra ones. These are some of the interfaces and some wireframes for some of them:

- A login page.
- A registration page.
- A food list (with all CRUD implementation, except for 'update').
- A recipes list (with all CRUD implementation, except for 'update').
- A general shopping list view (all missing food for all your recipes and total price).
- A list of all public recipes from other users with their names and total prices.

### Project requirements

- You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
- Login page and registration page:
    - Should be built with Devise.
- Food list:
    - Should display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also quantity of a given food).
    - Should lead to a form that allows users to add new food.
- Recipes list:
    - Should display a list of recipes created by the logged-in user as in the wireframe.
    - Should lead to recipe details.
    - If the user is the owner of the recipe, should allow the user to delete it.
- Public recipe list:
    - Should display a list of all public recipes ordered by newest as in the wireframe.
    - Should lead to recipe details.
- Recipe details:
    - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
    - If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- General shopping list view (**only for a group that has less than 3 members**):
    - Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
    - Should count the total food items and total price of the missing food.
- Make sure there are no N+1 queries happening.
- Create a navigation menu that allows users to open all of the pages you created.

### CODE REVIEW

File: spec/models/food_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Food, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (AnselemOdims) [inline spec/models/food_spec.rb:5]: - [ ] Please kindly include tests for this model

File: spec/models/recipe_food_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe RecipeFood, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (AnselemOdims) [inline spec/models/recipe_food_spec.rb:5]: - [ ] Please kindly include tests for this model

File: spec/models/recipe_spec.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Recipe, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (AnselemOdims) [inline spec/models/recipe_spec.rb:5]: - [ ] Please kindly include tests for this model

File: spec/requests/food_spec.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe 'Foods', type: :request do
+  describe 'GET /index' do
+    pending "add some examples (or delete) #{__FILE__}"
+  end
+end
```
Comment (AnselemOdims) [inline spec/requests/food_spec.rb:7]: - [ ] Please kindly include tests for this request

File: spec/requests/recipe_food_spec.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe 'RecipeFoods', type: :request do
+  describe 'GET /index' do
+    pending "add some examples (or delete) #{__FILE__}"
+  end
+end
```
Comment (AnselemOdims) [inline spec/requests/recipe_food_spec.rb:7]: - [ ]  Please kindly include tests for this request

File: spec/requests/recipe_spec.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe 'Recipes', type: :request do
+  describe 'GET /index' do
+    pending "add some examples (or delete) #{__FILE__}"
+  end
+end
```
Comment (AnselemOdims) [inline spec/requests/recipe_spec.rb:7]: - [ ] Please kindly include tests for this request

File: spec/requests/user_spec.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :request do
+  describe 'GET /index' do
+    pending "add some examples (or delete) #{__FILE__}"
+  end
+end
```
Comment (AnselemOdims) [inline spec/requests/user_spec.rb:7]: - [ ]  Please kindly include tests for this request

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @team, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/requests/user_spec.rb (hunk 7 above)
Comment (PRATAP-KUMAR) [reply spec/requests/user_spec.rb]: @AnselemOdims  Thank you very much for your review on our PR and good points to improve our RecipeApp, we have implemented the UI changes required and additional tests.




The tests for model specs and request specs

image

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @TEAM

Your project is complete! Great job for making the requested changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request✔️ 
- CRUD methods ok✔️ 
- test✔️ 
- Database ✔️ 
- Routes and pages ✔️ 
- No linter errors✔️ 

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

tufoin

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**